File: 2697. Lexicographically Smallest Palindrome.rb
#* 2697. Lexicographically Smallest Palindrome
s = "egcfe"
# @param {String} s
# @return {String}
def make_smallest_palindrome(s)
    p s
    p s.reverse
    p x = s.chars.zip(s.chars.reverse)
    x.map(&:min).join
end
p make_smallest_palindrome(s)

def make_smallest_palindrome(s)
  szip = s.chars.zip(s.chars.reverse)
  szip.map(&:min).join
end
p make_smallest_palindrome(s)

def make_smallest_palindrome(s)
  s.chars.zip(s.chars.reverse).map(&:min).join
end
p make_smallest_palindrome(s)

File: 2697. Lexicographically Smallest Palindrome.py
#* 2697. Lexicographically Smallest Palindrome
s = "egcfe"

class Solution:
    def makeSmallestPalindrome(self, s: str) -> str:
        x = list(zip(s, reversed(s)))
        return ''.join(min(pair) for pair in x)

print(Solution().makeSmallestPalindrome(s))

class Solution:
    def makeSmallestPalindrome(self, s: str) -> str:
        letters = list(s)
        print(letters)

        for i in range(len(s) // 2):
            letters[i] = letters[~i] = min(letters[i], letters[~i])

        return ''.join(letters)

print(Solution().makeSmallestPalindrome(s))